routing/routing.go

```diff
@@ -422,6 +422,11 @@ func (r *Routing) Get() *RouteLookup {
 
 // Close closes routing, stops receiving routes.
 func (r *Routing) Close() {
+	var rt *routeTable
+	rt, ok := r.routeTable.Load().(*routeTable)
+	if ok {
+		rt.Close()
+	}
```

Comment:
I think it is better to put it into startReceivingUpdates into `case <-r.quit:` - this way we know there will be no updates anymore

Reply:
It's not possible because routeTable is nil, just check the 2-3 lines above the `<-quit` case, there we set `rt = nil`

Reply:
I mean in the startReceivingUpdates, not in the receiveRouteMatcher

Reply:
changed, let's see what tests are saying. One problem could be that it's async closing by another goroutine and goroutine leak detection might fail